lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadataInfo.php

```
@@ -1181,7 +1189,20 @@ public function mapField(array $mapping)
             $mapping['nullable'] = false;
         }
 
-        if (isset($mapping['reference']) && ! empty($mapping['simple']) && ! isset($mapping['targetDocument'])) {
+        // Synchronize the "simple" and "storeAs" mapping information for backwards compatibility
+        if (isset($mapping['simple']) && ($mapping['simple'] === true || $mapping['simple'] === 'true')) {
+            $mapping['storeAs'] = ClassMetadataInfo::REFERENCE_STORE_AS_ID;
+        }
+        // Remove the "simple" mapping and use "storeAs" in all further logic
+        if (isset($mapping['simple'])) {
+            unset($mapping['simple']);
+        }
+
+        if (isset($mapping['reference'])
+            && isset($mapping['storeAs'])
+            && $mapping['storeAs'] === ClassMetadataInfo::REFERENCE_STORE_AS_ID
+            && ! isset($mapping['targetDocument'])
```

Not having a `targetDocument` set is one case where we need to throw an exception - the other would be `targetDocument` being stored in a different database than the document storing the reference.